src/ape/managers/config.py

```diff
@@ -83,6 +151,13 @@ def _plugin_configs(self) -> Dict[str, ConfigItem]:
         self.name = user_config.pop("name", "")
         self.version = user_config.pop("version", "")
 
+        dependencies = user_config.pop("dependencies", []) or []
+        if not isinstance(dependencies, list):
+            raise ConfigError("'dependencies' config item must be a list of dicts.")
+
+        # TODO: Handle decoding multiple dependency types
```

Comment:
This might be part of the work for getting multiple dependency APIs working together? We will have to make `ConfigManager` a `BaseModel`, which I think makes a lot of sense since it is configuration-file driven.

I will create a ticket for this once we are happy with this design, e.g. the new `ProjectAPI` + `DependencyAPI(ProjectAPI)`.

Reply:
It looks like pydantic does not support compound types yet for their smart union.